### PR TITLE
[sui core] check for effects before verifying cert

### PR DESCRIPTION
**Pull request by sblackshear**
I believe this will be needed after we have reconfig. The signature checking logic looks at the *current* committee, but this will make signature checking fail if you're looking at a cert w/ a different committee from a previous epoch.

In addition, this saves a signature check on the happy path when someone is asking for effects from a cert we've previously executed.

### CODE REVIEW

File: crates/sui-core/src/authority_server.rs [hunk 1]
```diff
@@ -389,7 +389,17 @@ impl ValidatorService {
...
+            .await
+            .map_err(|e| tonic::Status::internal(e.to_string()))?
+        {
+            return Ok(tonic::Response::new(response));
+        }
+
+        // 2) Verify cert signatures
         let cert_verif_metrics_guard = start_timer(metrics.cert_verification_latency.clone());
 
         certificate
```
Comment (sblackshear) [inline crates/sui-core/src/authority_server.rs:405]: Note: 396 is where we look at the current committee.

**Review by lxfind — APPROVED** [review]
Good catch.
I was just fixing a similar things from the other end (authority aggregator) in https://github.com/MystenLabs/sui/pull/5373

**Comment by lxfind** [comment]
Actually, hold on merging this PR. I have some questions, will discuss tomorrow.

**Review by kchalkias — APPROVED** [review]
Nice catch

**Comment by lxfind** [comment]
This is good for now, I think. Please feel free to merge.
For context, we had a discussion in https://mysten-labs.slack.com/archives/C02GD7J9HUM/p1666197929342289

**Pull request merged**